include/jsoncons/json.hpp

```diff
@@ -1129,679 +1083,84 @@ class basic_json
             return !(*this == rhs);
         }
 
-        template <class Alloc = allocator_type>
-        typename std::enable_if<std::is_pod<typename std::allocator_traits<Alloc>::pointer>::value,void>::type
-        swap(variant& other) JSONCONS_NOEXCEPT
-        {
-            if (this ==&other)
-            {
-                return;
+    private:
+        template <class TypeA, class TypeB>
+        void swap_a_b(variant& other)
+        {
+            TypeA& curA = as<TypeA>();
+            TypeB& curB = other.as<TypeB>();
+            TypeB tmpB(std::move(curB));
+            other.construct_var<TypeA>(std::move(curA));
+            construct_var<TypeB>(std::move(tmpB));
+        }
```

Comment:
The swap function was very huge. I simplified it into 3 template functions. I think that is easier to read and requires little maintenance.